src/plugins/mf_priority.cpp

```diff
@@ -0,0 +1,271 @@
+/************************************************************\
+ * Copyright 2021 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* mycpppriority.cpp - custom basic job priority plugin
+ *
+ */
+
+extern "C" {
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+}
+#include <flux/core.h>
+#include <flux/jobtap.h>
+#include <map>
+#include <iterator>
+#include <cmath>
+#include <cassert>
+#include <algorithm>
+#include <cinttypes>
+
+std::multimap<int, std::tuple<std::string, double>> users;
+
+/******************************************************************************
+ *                                                                            *
+ *                           Helper Functions                                 *
+ *                                                                            *
+ *****************************************************************************/
+
+/*
+ * Calculate a user's job priority using the following factors:
+ *
+ * fairshare: the ratio between the amount of resources allocated vs. resources
+ *     consumed.
+ * urgency: a user-controlled factor to prioritize their own jobs.
+ */
+int64_t priority_calculation (flux_plugin_t *p,
+                              flux_plugin_arg_t *args,
+                              int userid,
+                              char *bank,
+                              int urgency)
+{
+    double fshare_factor = 0.0, priority = 0.0;
+    int fshare_weight;
+    std::map<int, std::tuple<std::string, double>>::iterator it;
+
+    fshare_weight = 100000;
+
+    if (urgency == FLUX_JOB_URGENCY_HOLD)
+        return FLUX_JOB_PRIORITY_MIN;
+
+    if (urgency == FLUX_JOB_URGENCY_EXPEDITE)
+        return FLUX_JOB_PRIORITY_MAX;
+
+    // search element in multimap by key
+    it = users.find (userid);
+    if (it == users.end ())
+        return flux_jobtap_reject_job (p, args,
+                                       "user not found in flux-accounting DB");
```

Comment:
`flux_jobtap_reject_job()` can only be used from `job.validate` (or `job.dependency.*`), but `priority_calculation()` looks like it can be called from `job.priority.get`. In this case, you can return `flux_jobtap_priority_unavail()` if the job is still in PRIORITY state. It looks like this can't happen though -- if the user does not exist in the db then the job would have been rejected by `validate_cb`.

Reply:
Ah, yes I think you're right - `validate_cb ()` would have rejected the job first if the user does not exist in the DB, so I don't think there is a need for the `flux_jobtap_reject_job ()` in `priority_calculation ()`; I can remove that.